diff --git a/third_party/unrar/file.cpp b/third_party/unrar/file.cpp
--- a/third_party/unrar/file.cpp
+++ b/third_party/unrar/file.cpp
@@ -52,7 +52,7 @@ bool File::Open(const std::wstring &Name,uint Mode)
   uint Access=WriteMode ? GENERIC_WRITE:GENERIC_READ;
   if (UpdateMode)
     Access|=GENERIC_WRITE;
-  uint ShareMode=(Mode & FMF_OPENEXCLUSIVE) ? 0 : FILE_SHARE_READ;
+  uint ShareMode=(Mode & FMF_OPENEXCLUSIVE) ? 0 : FILE_SHARE_DELETE | FILE_SHARE_READ;
   if (OpenShared)
     ShareMode|=FILE_SHARE_WRITE;
   uint Flags=FILE_FLAG_SEQUENTIAL_SCAN;
